gcp/biganimal-csp-setup

```diff
@@ -0,0 +1,281 @@
+    echo "Downloaded ${CURRENT_PATH}/${template_file}"
+  else
+    echo "Error: failed to download ${CURRENT_PATH}/${template_file}"
+    exit 1
+  fi
+}
+
+create_iam_custom_role()
+{
+  # verify if a custom role with the same name is already existing
+  role_exists=$(gcloud iam roles list --project "$project" \
+    --filter "name:biganimal_${suffix_role_name}" --format json | jq '. | length')
+  if [ "$role_exists" -gt 0 ]; then
+    # a custom role with the same name already exists
+    echo -e "Updating GCP custom role ${GREEN}biganimal_${suffix_role_name}${NC} within GCP project ${GREEN}${project}${NC}..."
+
+    # updating the GCP custom role
+    custom_role_id=$(gcloud iam roles list --project "$project" \
+      --filter "name:biganimal_${suffix_role_name}" --format json | jq -r '.[].name')
+
+    gcloud iam roles update "${custom_role_id##*/}" --project "$project" \
+      --file "${CURRENT_PATH}/${template_file}" --quiet \
+      --format json > "${TMPDIR}"/biganimal_updatecustomrole_output.json
+  else
+    # creating the GCP custom role
+    echo -e "Creating GCP custom role ${GREEN}biganimal_${suffix_role_name}${NC} within GCP project ${GREEN}${project}${NC}..."
+    gcloud iam roles create "biganimal_${suffix_role_name}" --project "$project" \
+      --file "${CURRENT_PATH}/${template_file}" --quiet \
+      --format json > "${TMPDIR}"/biganimal_createcustomrole_output.json
+    custom_role_id=$(gcloud iam roles list --project "$project" \
+      --filter "name:biganimal_${suffix_role_name}" --format json | jq -r '.[].name')
+  fi
+}
+
+create_service_account()
+{
+  # verifying if a GCP service account with the same name is already existing
+  sa_exists=$(gcloud iam service-accounts list --project "$project" \
+    --filter "name:biganimal-${suffix_sa_name}" --format json | jq '. | length')
+  if [ "$sa_exists" -eq 0 ]; then
+    # the service account does not exists, going to create it
+    echo -e "Creating GCP service account ${GREEN}biganimal-${suffix_sa_name}${NC} within GCP project ${GREEN}${project}${NC}..."
+    gcloud iam service-accounts create "biganimal-${suffix_sa_name}" --project "$project" \
+      --display-name "biganimal-${suffix_sa_name}" --description "BigAnimal Service Account" \
+      --format json > "${TMPDIR}"/biganimal_createserviceaccount_output.json
+  else
+    # the service account already exists, going to use it
+    echo -e "GCP service account ${GREEN}biganimal-${suffix_sa_name}${NC} already exists within GCP project ${GREEN}${project}${NC}..."
+  fi
+  service_account_id=$(gcloud iam service-accounts list --project "$project" \
+    --filter "name:biganimal-${suffix_sa_name}" --format json | jq -r '.[].email')
+
+  # assigning the custom BigAnimal Role to the service account for the given GCP project
+  echo -e "Allowing GCP service account ${GREEN}biganimal-${suffix_sa_name}${NC} to use the GCP project ${GREEN}${project}${NC}..."
+  gcloud projects add-iam-policy-binding "$project" \
```

Comment:
What about the output if the role is already bound to the service account?

Reply:
it works even if this `iam-policy-binding` already exists, so if it is not already existing it gets added, and if it already exists it gets updated. I've already tested it on the dev project